section_vector_index.go

```diff
@@ -0,0 +1,289 @@
+//go:build densevector
+// +build densevector
+
+package zap
+
+import (
+	"encoding/binary"
+	"math"
+
+	"github.com/RoaringBitmap/roaring"
+	index "github.com/blevesearch/bleve_index_api"
+	faiss "github.com/blevesearch/go-faiss"
+)
+
+func init() {
+	segmentSections[sectionVectorIndex] = &vectorIndexSection{}
+}
+
+type vectorIndexSection struct {
+}
+
+func (v *vectorIndexSection) Process(opaque map[int]resetable, docNum uint64, field index.Field, fieldID uint16) {
+	if fieldID == math.MaxUint16 {
+		return
+	}
+
+	if vf, ok := field.(index.DenseVectorField); ok {
```

Comment:
Recommend refactoring DenseVectorField -> VectorField (as commented in bleve_index_api).